lib/matplotlib/pyplot.py

```diff
@@ -425,13 +429,18 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
         if get_backend().lower() == 'ps':
             dpi = 72
 
-        figManager = new_figure_manager(num, figsize=figsize,
-                                        dpi=dpi,
-                                        facecolor=facecolor,
-                                        edgecolor=edgecolor,
-                                        frameon=frameon,
-                                        FigureClass=FigureClass,
-                                        **kwargs)
+        if hasattr(_backend_mod, 'Window'):  # Can we use the MEP 27 code?
```

Comment:
@tacaswell how do you see this way of getting the new figure manager? 
Personally I have never liked the `new_figure_manager` and `new_figure_manager_given_figure` methods, but having the mechanics in two separate places seems even less natural than those two awkward methods